app/src/androidTest/kotlin/at/bitfire/davdroid/db/AppDatabaseMigrationsTest.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
+ */
+
+package at.bitfire.davdroid.db
+
+import android.content.Context
+import androidx.room.Room
+import androidx.room.testing.MigrationTestHelper
+import androidx.sqlite.db.SupportSQLiteDatabase
+import androidx.sqlite.db.framework.FrameworkSQLiteOpenHelperFactory
+import androidx.test.platform.app.InstrumentationRegistry
+import at.bitfire.davdroid.db.Collection.Companion.TYPE_CALENDAR
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Rule
+import org.junit.Test
+
+class AppDatabaseMigrationsTest {
+
+    private val TEST_DB = "test"
+
+    val context: Context = InstrumentationRegistry.getInstrumentation().targetContext
+
+    private val autoMigrationSpecs = AppDatabase.autoMigrationSpecs.map { it(context) }
+
+    @Rule
+    @JvmField
+    val helper = MigrationTestHelper(
+        InstrumentationRegistry.getInstrumentation(),
+        AppDatabase::class.java,
+        autoMigrationSpecs,
+        FrameworkSQLiteOpenHelperFactory()
+    )
+
+    /**
+     * Used for testing the migration process between two versions.
+     * @param fromVersion The version from which to start testing.
+     * @param toVersion The target version to test
+     * @param setup Run here all the SQL queries that prepare the database with the data required
+     * to perform the migration process.
+     * @param assertionsBlock Run here all the assertions for [AppDatabase] to make sure the
+     * migration was performed correctly.
+     */
+    private fun testMigration(
+        fromVersion: Int,
+        toVersion: Int,
+        setup: SupportSQLiteDatabase.() -> Unit,
+        assertionsBlock: (AppDatabase) -> Unit
+    ) {
+        var db = helper.createDatabase(TEST_DB, fromVersion).apply {
+            setup()
+
+            // Prepare for the next version.
+            close()
+        }
+
+        // Re-open the database with the new version and provide all the migrations.
+        db = helper.runMigrationsAndValidate(TEST_DB, toVersion, true, *AppDatabase.migrations)
+
+        // Once the migration process is complete, load AppDatabase
+        val database = Room.databaseBuilder(context, AppDatabase::class.java, TEST_DB)
+            // Include all manual migrations
+            .addMigrations(*AppDatabase.migrations)
+            // And all auto migration specs
+            .apply {
+                for (spec in autoMigrationSpecs) {
+                    addAutoMigrationSpec(spec)
+                }
+            }
+            .build()
+        assertionsBlock(database)
+    }
+
+    /**
+     * Test migrations from full VTIMEZONE to just timezone ID
+     */
+    @Test
+    fun migrate15To16_WithTimezone() {
```

Comment:
So these tests require API level 34, right? We should then use `@SdkSuppress` for the respective tests to avoid their execution on previous SDK levels (which may lead to CI failures etc). We should also note why level 34 is needed there.